packages/graphql-yoga/__tests__/subscriptions.spec.ts

```diff
@@ -86,4 +85,58 @@ describe('Subscription', () => {
       }
     }
   })
+
+  test('should issue pings while connected', async () => {
+    const schema = createSchema({
+      typeDefs: /* GraphQL */ `
+        type Subscription {
+          hi: String!
+        }
+        type Query {
+          hi: String!
+        }
+      `,
+      resolvers: {
+        Subscription: {
+          hi: {
+            async *subscribe() {
+              await new Promise((resolve) => setTimeout(resolve, 3000))
+              yield { hi: 'hi' }
+            },
+          },
+        },
+      },
+    })
+
+    const yoga = createYoga({ schema })
+
+    const response = await yoga.fetch('http://yoga/graphql', {
+      method: 'POST',
+      headers: {
+        'content-type': 'application/json',
+        accept: 'text/event-stream',
+      },
+      body: JSON.stringify({
+        query: /* GraphQL */ `
+          subscription {
+            hi
+          }
+        `,
+      }),
+    })
+
+    let body = ''
+    for await (const chunk of response.body!) {
+      body += chunk.toString()
+    }
+    expect(body).toMatchInlineSnapshot(`
```

Comment:
You can also try `await response.text()`

Reply:
I tried, but `.text()` resolves immediately before the iterator completes. And now I realised I was missing an `await`. 😓 

https://github.com/dotansimha/graphql-yoga/pull/2410/commits/2240010bb3853da426d46059de74538d60031f76